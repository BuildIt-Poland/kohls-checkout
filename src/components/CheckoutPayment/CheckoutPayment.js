import React from 'react';

function CheckoutPayment() {
  return (
    <section>
      <h3>CheckoutPayment placeholder</h3>
    </section>
  );
}

export default CheckoutPayment;
