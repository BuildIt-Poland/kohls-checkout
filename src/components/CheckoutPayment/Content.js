import styled from 'styled-components';

import { spacingLarge } from '../../styles/designTokens';
import ContentBase from '../Content';

const Content = styled(ContentBase)`
  padding-bottom: ${spacingLarge};
`;

export default Content;
