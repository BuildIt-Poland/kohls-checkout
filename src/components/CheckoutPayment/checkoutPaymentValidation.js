import * as Yup from 'yup';

const checkoutPaymentValidation = Yup.object().shape({
  cardHolderName: Yup.string()
    .min(3, 'Too short!')
    .required('Required'),
  cardNumber: Yup.string()
    .matches(/^[^A-Z^a-z]*$/, 'Only numbers allowed')
    .min(13, 'Too short!')
    .max(16, 'Too long!')
    .required('Required'),
  expMonth: Yup.string()
    .matches(/^((?!MM).)*$/, 'Required')
    .required('Required'),
  expYear: Yup.string()
    .matches(/^((?!YY).)*$/, 'Required')
    .required('Required'),
  securityCode: Yup.string()
    .min(3, 'Too short!')
    .max(4, 'Too long!')
    .required('Required')
});

export default checkoutPaymentValidation;
