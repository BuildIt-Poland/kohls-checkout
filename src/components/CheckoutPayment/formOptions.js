// Generates options for mont and year expiry selects

export function months() {
  return [...Array(12)].map((_, x) => {
    // Adds leading zero to months < 10
    return String(++x).padStart(2, 0);
  });
}

export function nextYearsFromDate(startDate, yearsAmount = 20) {
  const startYear = Number(new Date(startDate).getFullYear());

  // Leaves only 2 last year numbers
  return [...Array(yearsAmount + 1)].map((_, x) => String(x + startYear).slice(-2));

  // Shows full year
  // return [...Array(yearsAmount + 1)].map((_, x) => x + startYear);
}

export function selectOptions(values, defaultOption = []) {
  return defaultOption.concat(
    values.map(value => {
      return {
        label: value,
        value
      };
    })
  );
}

export function monthsOptions() {
  const defaultOption = [{ label: 'MM', value: 'MM', disabled: true }];
  const rawMonths = months();

  return selectOptions(rawMonths, defaultOption);
}

export function yearsOptions(startDate, yearsAmount) {
  const defaultOption = [{ label: 'YY', value: 'YY', disabled: true }];
  const rawYears = nextYearsFromDate(startDate, yearsAmount);

  return selectOptions(rawYears, defaultOption);
}
