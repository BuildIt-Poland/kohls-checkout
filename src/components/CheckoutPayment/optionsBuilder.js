/**
 * Generate Options for Months checkbox
 *
 * @param {Number} numberOfMonths - Amount of months to generate options. (12 by default)
 */
const generateMonths = (numberOfMonths = 12) => {
  const monthOptions = [{ label: 'MM', value: 'MM', disabled: true }];

  for (let i = 1; i <= numberOfMonths; i++) {
    monthOptions.push({
      label: `${i}`,
      value: `${i}`
    });
  }
  return monthOptions;
};

/**
 * Generate Options for Years checkbox
 *
 * @param {Number} yearsAmount - Amount of years to generate options. (20 by default)
 */
const generateYears = (yearsAmount = 20) => {
  const yearOptions = [{ label: 'YY', value: 'YY', disabled: true }];

  let currentYear = Number(
    new Date()
      .getFullYear()
      .toString()
      .slice(2)
  );

  for (let i = 1; i <= yearsAmount; i++) {
    yearOptions.push({
      label: `${currentYear}`,
      value: `${currentYear}`
    });
    currentYear++;
  }

  return yearOptions;
};

export { generateMonths, generateYears };
