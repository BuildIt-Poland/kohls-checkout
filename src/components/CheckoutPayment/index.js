import { compose } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { setPaymentInfo } from '../../actions/paymentInfoActions';
import CheckoutPayment from './CheckoutPayment';

function mapStateToProps(state) {
  return {
    initialPaymentInfo: state.paymentInfo
  };
}

function mapDispatchToProps(dispatch) {
  return {
    setPaymentInfo: info => dispatch(setPaymentInfo(info))
  };
}

export default compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(CheckoutPayment);
