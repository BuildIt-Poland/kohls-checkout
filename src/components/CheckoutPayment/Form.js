import React from 'react';
import { Form as FormikForm } from 'formik';

import Text from '../Text';
import FormCreditCardInput from '../FormCreditCardInput';
import FormInput from '../FormInput';
import FormSelect from '../FormSelect';
import { generateMonths, generateYears } from './optionsBuilder';

const expMonthOptions = generateMonths();
const expYearsOptions = generateYears();

function Form() {
  return (
    <FormikForm>
      <Text dimmed>Please complete all fields.</Text>
      <FormInput name="cardHolderName" label="Cardholder Name" />
      <FormCreditCardInput name="cardNumber" label="Card Number" maxLength="19" />
      <FormSelect name="expMonth" label="Exp. Month" options={expMonthOptions} />
      <FormSelect name="expYear" label="Exp. Year" options={expYearsOptions} />
      <FormInput name="securityCode" type="password" label="Security Code" maxLength="4" />
    </FormikForm>
  );
}

export default Form;
