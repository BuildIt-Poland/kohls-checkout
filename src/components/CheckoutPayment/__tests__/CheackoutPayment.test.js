import React from 'react';
import renderer from 'react-test-renderer';

import CheckoutPayment from '../CheckoutPayment';

jest.mock('react-router-dom', () => ({
  Link: 'Link'
}));

describe('COMPONENT - CheckoutPayment', () => {
  it('renders CheckoutPayment component', () => {
    const component = renderer.create(<CheckoutPayment />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
