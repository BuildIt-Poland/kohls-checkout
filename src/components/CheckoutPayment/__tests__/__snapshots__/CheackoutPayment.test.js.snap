// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - CheckoutPayment renders CheckoutPayment component 1`] = `
Array [
  .c3 {
  font-size: 1.2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: normal;
}

.c6 {
  font-size: 1.2rem;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: normal;
}

.c0 {
  font-size: 1.8rem;
  text-align: center;
  margin: 10px;
}

.c5 {
  font-size: 1.2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: bold;
  color: #9cc472;
}

.c7 {
  font-size: 1.2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: normal;
  color: #9cc472;
}

.c4 {
  float: right;
}

.c2 {
  margin: 0;
}

.c8 {
  font-size: 1.2rem;
  font-weight: bold;
}

.c1 {
  padding-left: 2rem;
  padding-right: 2rem;
}

<section>
    <h3
      className="c0"
    >
      Order Summary
    </h3>
    <div
      className="c1"
    >
      <p
        className="c2"
      >
        <span
          className="c3"
        >
          Subtotal
          <span
            className="c4"
          >
            $125.00
          </span>
        </span>
      </p>
      <p
        className="c2"
      >
        <span
          className="c5"
        >
          Kohl's Cash & Discounts
          <span
            className="c4"
          >
            -$18.75
          </span>
        </span>
      </p>
      <p
        className="c2"
      >
        <span
          className="c6"
        >
          Shipping
          <span
            className="c4"
          >
            Free
          </span>
        </span>
      </p>
      <span
        className="c7"
      >
        You have earned free shipping.
      </span>
      <p
        className="c2"
      >
        <span
          className="c6"
        >
          Tax
          <span
            className="c4"
          >
            $6.38
          </span>
        </span>
      </p>
      <hr />
      <p
        className="c2"
      >
        <span
          className="c8"
        >
          TOTAL
          <span
            className="c4"
          >
            $112.63
          </span>
        </span>
      </p>
      <p
        className="c2"
      >
        <span
          className="c3"
        >
          Your Savings
          <span
            className="c4"
          >
            $122.70
          </span>
        </span>
      </p>
    </div>
  </section>,
  .c1 {
  background: #00a09f;
  color: #ffffff;
  display: block;
  font-size: 1.8rem;
  font-weight: 600;
  padding: 1.2rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
}

.c0 {
  bottom: 0;
  left: 0;
  padding: 1.2rem;
  position: fixed;
  width: 100%;
  box-shadow: 0 0 5px rgba(0,0,0,0.5);
}

<div
    className="c0"
  >
    <Link
      className="c1"
      to="/checkout/review"
    >
      Review Order
    </Link>
  </div>,
]
`;
